### PR TITLE
fix: Quote profile json when passing to kando in go code

### PR DESCRIPTION
**Pull request by hairyhum**
## Change Overview

Add quoting when building kando command with profile in go code.
Add unit test case for non-null profile.

Fixes https://github.com/kanisterio/kanister/security/code-scanning/198 and https://github.com/kanisterio/kanister/security/code-scanning/199

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [x] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test
- [ ] :building_construction: Build

## Issues 

- fixes #issue-number

## Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/function/rds_functions_test.go [hunk 1]
```diff
@@ -106,7 +128,7 @@ func (s *RDSFunctionsTest) TestPrepareCommand(c *check.C) {
 			for db in "${dblist[@]}";
 			  do echo "backing up $db db" && pg_dump $db -C --inserts > /backup/$db.sql;
 			done
-			tar -zc backup | kando location push --profile '%s' --path "${BACKUP_PREFIX}/${BACKUP_ID}" -
+			tar -zc backup | kando location push --profile "%s" --path "${BACKUP_PREFIX}/${BACKUP_ID}" -
```
Comment (viveksinghggits) [inline pkg/function/rds_functions_test.go:131]: I see that at some places we are providing profile in double quotes `""` and at some place without. Is that intentional?

**Review by viveksinghggits — APPROVED** [review]

File: pkg/function/rds_functions_test.go (hunk 1 above)
Comment (hairyhum) [reply pkg/function/rds_functions_test.go:131]: In some tests we just pass `null` string to this function. We can make it more consistent by passing `"\"null\""` instead, I guess.

Comment (viveksinghggits) [reply pkg/function/rds_functions_test.go:131]: I am good, even if you want to merge this with current state.
